src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Rendering/EventForDotNet.ts

```diff
@@ -69,18 +74,137 @@
       case 'pointerout':
       case 'pointerover':
       case 'pointerup':
-        return new EventForDotNet<UIPointerEventArgs>('pointer', { Type: event.type });
+        return new EventForDotNet<UIPointerEventArgs>('pointer', parsePointerEvent(event));
 
+      case 'wheel':
       case 'mousewheel':
-        return new EventForDotNet<UIWheelEventArgs>('wheel', { Type: event.type });
-
+        return new EventForDotNet<UIWheelEventArgs>('wheel', parseWheelEvent(event));
 
       default:
-        return new EventForDotNet<UIEventArgs>('unknown', { Type: event.type });
+        return new EventForDotNet<UIEventArgs>('unknown', { type: event.type });
     }
   }
 }
 
+function parseDragEvent(event: any) {
+  return {
+    type: event.type,
+    detail: event.detail,
+    dataTransfer: event.dataTransfer,
+    screenX: event.screenX,
+    screenY: event.screenY,
+    clientX: event.clientX,
+    clientY: event.clientY,
+    button: event.button,
+    buttons: event.buttons,
+    mozPressure: event.mozPressure,
+    ctrlKey: event.ctrlKey,
+    shiftKey: event.shiftKey,
+    altKey: event.altKey,
+    metaKey: event.metaKey
+  }
+}
+
+function parseWheelEvent(event: any) {
+  return Object.assign(parseMouseEvent(event),
+    {
+      deltaX: event.deltaX,
+      deltaY: event.deltaY,
+      deltaZ: event.deltaZ,
+      deltaMode: event.deltaMode
+    });
+}
+
+function parseProgressEvent(event: any) {
+  return {
+    type: event.type,
+    lengthComputable: event.legthComputable,
+    loaded: event.loaded,
+    total: event.total
+  };
+}
+
+function parseTouchEvent(event: any) {
+
+  function parseTouch(touchList: TouchList) {
+    const touches: UITouchPoint[] = [];
```

Comment:
Stylistic point: when mapping arrays in TypeScript, can we use `.map` instead of a `for` loop? This function can be reduced to pretty much a one-liner.

Reply:
Yeah I know... But unfortunately, the `TouchList` type can't be iterated :(

We have to call `tl.item(idx)`... Weird, but yeah, that is the way...

Reply:
Ah OK, thanks for clarifying!